ddl/ddl_api.go

```diff
@@ -1054,7 +1063,15 @@ func checkDuplicateConstraint(namesMap map[string]bool, name string, foreign boo
 
 func setEmptyConstraintName(namesMap map[string]bool, constr *ast.Constraint, foreign bool) {
 	if constr.Name == "" && len(constr.Keys) > 0 {
-		colName := constr.Keys[0].Column.Name.L
+		var colName string
+		for _, keyPart := range constr.Keys {
+			if keyPart.Expr != nil {
+				colName = "expression_index"
+			}
```

Comment:
Will this work in the case that there are two expression indexes on the same table? I checked, and there are currently no tests for confirming two expression indexes will work.

Reply:
I think it works.
```
mysql> create table t2(a int, key((a+1)), key((a+2)), key idx((a+3)), key((a+4)));
Query OK, 0 rows affected (0.01 sec)

mysql> show create table t2;
+-------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                                                                                                                                                              |
+-------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| t2    | CREATE TABLE `t2` (
  `a` int(11) DEFAULT NULL,
  KEY `expression_index` ((`a` + 1)),
  KEY `expression_index_2` ((`a` + 2)),
  KEY `idx` ((`a` + 3)),
  KEY `expression_index_3` ((`a` + 4))
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin |
+-------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```
Anyway, I will add a test cover that later.

Reply:
Why not name it `functional_index` as MySQL does?

Reply:
Following the previous design. Since we use the term `expression index`, but not `function index`.